media/media.go

```diff
@@ -41,12 +48,18 @@ func loadImageFile(path string) (image.Image, error) {
 	return img, err
 }
 
-func imageCacheKey(path string, w, h int) string {
-	return fmt.Sprintf("%s:%d:%d", path, w, h)
+// imageCacheKey creates a cache key that is unique to the given parameters.
+// color can be nil when there is no color.
+func imageCacheKey(path string, w, h int, c color.Color) string {
+	return fmt.Sprintf("%s:%d:%d:%s", path, w, h, colorToHex6(c))
 }
 
+// loadLocalImageAsset loads the specified media asset from the local filesystem.
+// It assumes the "mediaPath" is relative to the base path.
+// The base path is the directory where all media assets should be located within.
 func loadLocalImageAsset(mediaPath string, w, h int) (image.Image, error) {
-	key := imageCacheKey(mediaPath, w, h)
+	key := imageCacheKey(mediaPath, w, h, nil)
+	mediaPath = filepath.Join(baseDir, mediaPath) // file is within the baseDir on disk
```

Comment:
Silly question - any idea if go compiler will re-order to happen after the cache check? Since it's technically not needed this early.

Reply:
I don't know :eyes: